server.js

```diff
@@ -3,36 +3,130 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+
+import topMusicData from './data/top-music.json'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+// Defines the port the app will run on.
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+
+const Record = new mongoose.model('Record', {
+  id: Number,
+  trackName:String,
+  artistName:String,
+  genre: String,
+  bpm: Number,
+  energy: Number,
+  danceability: Number,
+  loudness: Number,
+  liveness: Number,
+  valence: Number,
+  length: Number,
+  acousticness: Number,
+  speechiness: Number,
+  popularity: Number
+});
+
+//Populating database
+
+if(process.env.RESET_DATABASE) {
+  const populateDatabase = async () => {
+    await Record.deleteMany();
+    topMusicData.forEach(async item => {
+      await new Record(item).save();
+    })
+  }
+  populateDatabase();
+}
+
+//Provide error of server
+
+app.use((req, res, next) => {
+if(mongoose.connection.readyState === 1) {
+  next()
+} else {
+  res.status(503).json({error: "Service unavailable at the moment"})
+}
 })
 
+app.get('/', (req, res) => {
+    res.send('Hello, welcome to this music records api built with MongoDB')
+  })
+
+//Find all records and query by every attribute and a combination of them
+  app.get('/records', async (req, res) => {
+  const allRecords = await Record.find(req.query);
+    res.json(allRecords)
+  })
+
+  //Limiting returning data
+app.get('/records20', async (req, res) => {
+  const allRecords = await Record.find(req.query).skip(20).limit(20);
+    res.json(allRecords)
+  })
+
+  //Route to find one specific record by id
+app.get('/records/id/:id', async (req, res) => {
```

Comment:
`id` is a default key for a document, so proper endpoint would be  `/records/:id`